### PR TITLE
Use https links

### PR DESCRIPTION
**Pull request by l2dy**
`find ./* -name '*.php' | xargs sed -i 's|href="http://|href="https://|g'`

### CODE REVIEW

**Comment by kecrily** [comment]
duplicate https://github.com/typecho/typecho/pull/1006 and seeing joyqi's reply.

**Comment by l2dy** [comment]
> duplicate #1006 and seeing joyqi's reply.

i18n files could have been updated with https://github.com/typecho/languages/pull/48. Any other reason it was not merged?

I could revert changes to links in `_e(...)` if i18n is still of concern.

**Comment by joyqi** [comment]
This pr will be merged in v1.2.1.

**Comment by l2dy** [comment]
Fixed merge conflict. @joyqi Could you merge this now?

**Comment by l2dy** [comment]
@joyqi Hi, could you merge this before merge conflicts happen again?

File: var/Typecho/Common.php [hunk 1]
```diff
@@ -417,10 +417,10 @@ public static function fixHtml(?string $string): ?string
          * 去掉字符串中的html标签
          * 使用方法:
          * <code>
-         * $input = '<a href="http://test/test.php" title="example">hello</a>';
+         * $input = '<a href="https://test/test.php" title="example">hello</a>';
          * $output = Common::stripTags($input, <a href="">);
          * echo $output;
-         * //display: '<a href="http://test/test.php">hello</a>'
+         * //display: '<a href="https://test/test.php">hello</a>'
```
Comment (sy-records) [inline var/Typecho/Common.php:423]: 这的可以不用改

Comment (l2dy) [reply var/Typecho/Common.php]: Fixed.

File: var/Widget/Options/Discussion.php [hunk 2]
```diff
@@ -127,7 +127,7 @@ public function form(): Form
             $this->options->commentDateFormat,
             _t('评论日期格式'),
             _t('这是一个默认的格式,当你在模板中调用显示评论日期方法时, 如果没有指定日期格式, 将按照此格式输出.') . '<br />'
-            . _t('具体写法请参考 <a href="http://www.php.net/manual/zh/function.date.php">PHP 日期格式写法</a>.')
+            . _t('具体写法请参考 <a href="https://www.php.net/manual/zh/function.date.php">PHP 日期格式写法</a>.')
```
Comment (sy-records) [inline var/Widget/Options/Discussion.php:130]:
```suggestion
            . _t('具体写法请参考 <a href="https://www.php.net/date">PHP 日期格式写法</a>.')
```

File: var/Widget/Options/Reading.php [hunk 3]
```diff
@@ -105,7 +105,7 @@ public function form(): Form
             _t('文章日期格式'),
             _t('此格式用于指定显示在文章归档中的日期默认显示格式.') . '<br />'
             . _t('在某些主题中这个格式可能不会生效, 因为主题作者可以自定义日期格式.') . '<br />'
-            . _t('请参考 <a href="http://www.php.net/manual/zh/function.date.php">PHP 日期格式写法</a>.')
+            . _t('请参考 <a href="https://www.php.net/manual/zh/function.date.php">PHP 日期格式写法</a>.')
```
Comment (sy-records) [inline var/Widget/Options/Reading.php:108]:
```suggestion
            . _t('请参考 <a href="https://www.php.net/date">PHP 日期格式写法</a>.')
```

File: usr/themes/default/footer.php [hunk 4]
```diff
@@ -6,7 +6,7 @@
 
 <footer id="footer" role="contentinfo">
     &copy; <?php echo date('Y'); ?> <a href="<?php $this->options->siteUrl(); ?>"><?php $this->options->title(); ?></a>.
-    <?php _e('由 <a href="http://www.typecho.org">Typecho</a> 强力驱动'); ?>.
+    <?php _e('由 <a href="https://www.typecho.org">Typecho</a> 强力驱动'); ?>.
```
Comment (sy-records) [inline usr/themes/default/footer.php:9]:
```suggestion
    <?php _e('由 <a href="https://typecho.org">Typecho</a> 强力驱动'); ?>.
```

File: usr/themes/default/sidebar.php [hunk 5]
```diff
@@ -55,7 +55,7 @@
                 <?php endif; ?>
                 <li><a href="<?php $this->options->feedUrl(); ?>"><?php _e('文章 RSS'); ?></a></li>
                 <li><a href="<?php $this->options->commentsFeedUrl(); ?>"><?php _e('评论 RSS'); ?></a></li>
-                <li><a href="http://www.typecho.org">Typecho</a></li>
+                <li><a href="https://www.typecho.org">Typecho</a></li>
```
Comment (sy-records) [inline usr/themes/default/sidebar.php:58]:
```suggestion
                <li><a href="https://typecho.org">Typecho</a></li>
```

@joyqi typecho网站的重定向有些问题，`https://www.typecho.org` 会重定向到 http://typecho.org

**Review by sy-records — CHANGES_REQUESTED** [review]

File: var/Widget/Options/Discussion.php (hunk 2 above)
Comment (l2dy) [reply var/Widget/Options/Discussion.php:130]: What is this for?

Comment (sy-records) [reply var/Widget/Options/Discussion.php:130]: Let the php document choose its own language instead of forcing the Chinese document.

Comment (l2dy) [reply var/Widget/Options/Discussion.php:130]: I think it's out of scope for this PR, but I'll let the maintainers decide and have enabled "allow edits by maintainers".

File: usr/themes/default/sidebar.php (hunk 5 above)
Comment (joyqi) [reply usr/themes/default/sidebar.php]: > @joyqi typecho网站的重定向有些问题，`https://www.typecho.org` 会重定向到 http://typecho.org

已修复

**Pull request merged**